src/Kernel.php

```diff
@@ -30,68 +30,62 @@
  * builds a dispatcher to return a function that receives a messages and return the state
  *
  * usage:
- * $dispatch = buildDispatcher($eventStoreFactory, $snapshotStoreFactory, $commandMap, $producerFactory);
+ * $dispatch = buildDispatcher($eventStoreFactory, $snapshotStoreFactory, $aggregateDefinition, $handlerMap, $producerFactory);
  * $state = $dispatch($message);
  *
  * $producerFactory is expected to be a callback that returns an instance of Prooph\ServiceBus\Async\MessageProducer.
  * $commandMap is expected to be an array like this:
  * [
- *     RegisterUser::class => [
- *         'handler' => function (array $state, Message $message) use (&$factories): AggregateResult {
- *             return \Prooph\MicroExample\Model\User\registerUser($state, $message, $factories['emailGuard']());
- *         },
- *         'definition' => UserAggregateDefinition::class,
- *     ],
- *     ChangeUserName::class => [
- *         'handler' => '\Prooph\MicroExample\Model\User\changeUserName',
- *         'definition' => UserAggregateDefinition::class,
- *     ],
+ *     RegisterUser::class => function (array $state, Message $message) use (&$factories): AggregateResult {
+ *         return \Prooph\MicroExample\Model\User\registerUser($state, $message, $factories['emailGuard']());
+ *     },
+ *     ChangeUserName::class => \Prooph\MicroExample\Model\User\changeUserName,
  * ]
  * $message is expected to be an instance of Prooph\Common\Messaging\Message
  */
 function buildCommandDispatcher(
     callable $eventStoreFactory,
     callable $snapshotStoreFactory,
-    array $commandMap,
+    AggregateDefiniton $aggregateDefiniton,
```

Comment:
I thought like "microservice" - only one aggregate type per microservice is allowed.